drivers/input/input_gpio_qdec.c

```diff
@@ -157,6 +162,12 @@ static void gpio_qdec_sample_timer_timeout(struct k_timer *timer)
 	unsigned int key;
 	uint8_t step;
 
+#ifdef CONFIG_PM_DEVICE
+	if (atomic_get(&data->suspended) == 1) {
+		return;
+	}
```

Comment:
is local state necessary (data->suspended)?

Reply:
My reasoning is that without it, the timer could fire and interrupt mid-execution of M_DEVICE_ACTION_SUSPEND and reschedule the idle-work, which would partially re-init the device, and that with this check and `k_work_cancel_delayable_sync` after that the driver should be dead for sure. Does that make sense? The synchronization here is a bit tricky